### PR TITLE
Create a new page in devdocs for Accessibility

### PR DESCRIPTION
**Pull request by ryelle**
This PR is to start a discussion on what accessibility should mean in calypso, and what references we should add in our devdocs. I've copied @davidakennedy's resources for themers as a start, but I think it would be better to trim this down to only a few links per section.

So, what should "accessible" mean for calypso? Should we aim for [WCAG 2.0](https://www.w3.org/WAI/WCAG20/glance/Overview)? Maybe we can prioritize parts of that which would be easier to test for and implement?

Once we've decided that, we can tailor our references to working with and testing those aspects.

View the page here: https://calypso.live/devdocs/docs/accessibility.md?branch=add/accessibility-docs

### CODE REVIEW

**Comment by ryelle** [comment]
It looks like @drw158 has been targeting WCAG 2.0 AA with his work on text contrast #11464

**Comment by davidakennedy** [comment]
Thanks a lot for starting this conversation @ryelle!

I have a few thoughts:

* I'd love to see it stay simple and broad. Focus on the basics on this main page. Getting the basics right can help a lot. WordPress has a good [page on this](https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/), recently added to the core handbook. We could start there, and simplify that.
* Some of the resources, plus a few tips might make sense to add to individual component docs. Some patterns can have gotchas, and having a few pointers baked in might help them get observed and learned better.
* We don't have to start out with WCAG 2.0 as the mark to hit. That is the mark we should aim for in the long run, but I also think there's nothing wrong with starting smaller and building up. Say, really focusing on keyboard navigation/focus management. Then layer on more as checking for keyboard accessibility gets easier for most people involved.

I'll try to draft up something as a starting point at some point next week.

**Comment by ryelle** [comment]
> We don't have to start out with WCAG 2.0 as the mark to hit. That is the mark we should aim for in the long run, but I also think there's nothing wrong with starting smaller and building up.

Totally. I was hoping this doc would be a bit of both: a description of the long term goal, and some "requirements" that we want to meet immediately - keyboard/focus being a great start.

> Some of the resources, plus a few tips might make sense to add to individual component docs. 

That's a great idea; it reminds me of the [Spinner docs](https://wpcalypso.wordpress.com/devdocs/design/spinner) which specifically asks you to think twice about why you need a spinner.

**Comment by alisterscott** [comment]
Closing as this PR is stale - please reopen if necessary

**Comment by codebykat** [comment]
Gonna reopen and revisit this

**Comment by codebykat** [comment]
I cleaned up `Accessibility.md` a bit, though I think it still needs some work (I agree that it should probably have fewer things). Additionally, I added a more task-focused checklist / testing workflow that complements the merge checklist: https://calypso.live/devdocs/docs/accessibility-checklist.md?branch=add/accessibility-docs

**Comment by codebykat** [comment]
Github won't let me request a review from @ryelle because I stole this PR from her, hi Kelly

**Comment by codebykat** [comment]
I was inspired by a number of sources including:

https://davidakennedy.com/2014/09/12/a-workflow-for-testing-web-accessibility-with-free-tools/
WP a11y testing checklist: https://make.wordpress.org/themes/handbook/review/accessibility/required/ and https://codex.wordpress.org/Accessibility
http://accessibility.voxmedia.com/
https://a11yproject.com/checklist.html

Not sure if we want to put these links in the larger doc too (if so it's not obvious to me where to put them)

File: docs/accessibility.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+* [A11y Buzz](http://www.a11ybuzz.com/): Several web folks in the accessibility community launched this site in January 2012. It's a collection of accessibility resources that are triaged and reviewed by experts before they're posted.
+* [The Six Simplest Accessibility Tests Anyone Can Do](http://www.karlgroves.com/2013/09/05/the-6-simplest-web-accessibility-tests-anyone-can-do/): A handful of simple things you can do to think about and implement accessibility with little effort as you work on projects.
+* [Why Keyboard Accessibility Isn’t the Same Thing as Screen Reader Accessibility](https://www.ssbbartgroup.com/blog/why-keyboard-accessibility-isnt-the-same-thing-as-screen-reader-accessibility/): This blog post explains the three critical aspects of accessibility for interactive web technologies: Keyboard Accessibility, Screen Reader Accessibility and Cognitive Accessibility, and the differences between each.
+* [Teaching Accessibility Core Rules](http://www.deque.com/pragmatica11y-teaching-accessibility-core-rules): This post breaks down accessibility principles to their core, making them even easier to think about during projects.
```
Comment (ryelle) [inline docs/accessibility.md]: This link doesn't seem to exist anymore

File: docs/accessibility.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+* [WebAIM](http://webaim.org/): WebAIM has provided comprehensive web accessibility solutions since 1999, and is one of the leading providers of web accessibility expertise internationally. WebAIM is a non-profit organization within the [Center for Persons with Disabilities](http://www.cpd.usu.edu/) at [Utah State University](http://www.usu.edu/). The site brings together a huge collection of information about web accessibility. Most of it is up to date and evergreen enough to help you. The [organization's blog](http://webaim.org/blog/) tackles many modern-day, evolving topics related to web accessibility.
+* [W3C’s Web Accessibility Initiative](http://www.w3.org/WAI/): The World Wide Web Consortium's Web Accessibility Initiative provides strategies, guidelines and resources to make the Web accessible to people with disabilities.
+* [A11y Buzz](http://www.a11ybuzz.com/): Several web folks in the accessibility community launched this site in January 2012. It's a collection of accessibility resources that are triaged and reviewed by experts before they're posted.
```
Comment (ryelle) [inline docs/accessibility.md]: I think this is a different site now 🙁

File: docs/accessibility.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+* [The Six Simplest Accessibility Tests Anyone Can Do](http://www.karlgroves.com/2013/09/05/the-6-simplest-web-accessibility-tests-anyone-can-do/): A handful of simple things you can do to think about and implement accessibility with little effort as you work on projects.
+* [Why Keyboard Accessibility Isn’t the Same Thing as Screen Reader Accessibility](https://www.ssbbartgroup.com/blog/why-keyboard-accessibility-isnt-the-same-thing-as-screen-reader-accessibility/): This blog post explains the three critical aspects of accessibility for interactive web technologies: Keyboard Accessibility, Screen Reader Accessibility and Cognitive Accessibility, and the differences between each.
```
Comment (ryelle) [inline docs/accessibility.md]: Hm, I think we could have a different post here, since the previous link touches on keyboard a11y - maybe this [Intro to Cognitive Accessibility](http://jkg3.com/Journal/cognitive-accessibility-101-part-1-what-is-cognitive-accessibility).

File: docs/accessibility.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+* [Teaching Accessibility Core Rules](http://www.deque.com/pragmatica11y-teaching-accessibility-core-rules): This post breaks down accessibility principles to their core, making them even easier to think about during projects.
+* Presentation: [Explorations in the Virtual DOM: How React.js Impacts Accessibility](https://marcysutton.github.io/react-a11y-presentation/)
+* [Inclusive Design Fundamentals](https://isner.github.io/inclusive-design-fundamentals/), especially the [disability types](https://isner.github.io/inclusive-design-fundamentals/handouts/disability-types.html) to learn about how different disabilities affect how people use technology.
+
```
Comment (ryelle) [inline docs/accessibility.md]: Let's add the [Inclusive Components project](https://inclusive-components.design/) to this list too, they have a growing list of component patterns for things like menus, tooltips, etc along with information on why to use what ARIA properties.

File: docs/accessibility.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+### Going in depth
+
+A set of blogs, written by accessibility consultants and evangelists that go in depth on all things accessibility.
+
+* [The Paciello Group Blog](http://www.paciellogroup.com/blog/): A blog about web accessibility by one of the first accessibility consulting firms.
+* [Simply Accessible](http://simplyaccessible.com/): A blog with in-depth web accessibility tutorials.
+* [WebAxe](http://www.webaxe.org/): A blog and podcast about web accessibility.
+* [456 Berea Street](http://www.456bereastreet.com/): A blog on accessibility and web development by Roger Johansson.
```
Comment (ryelle) [inline docs/accessibility.md]: This blog hasn't been updated since 2016... not that the info is stale, but maybe we should bump this off the list?

File: docs/accessibility.md [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+* [Accessible Culture](http://www.accessibleculture.org/): A blog about web accessibility by Jason Kiss, an accessibility and standards researcher located in Wellington, New Zealand.
```
Comment (ryelle) [inline docs/accessibility.md]: This is also an inactive blog, so same question. IIRC I left these on the list for "extra resources" but I don't want to overload the page with links, it's pretty link-heavy already.

File: docs/accessibility.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+* [aXe](https://www.deque.com/products/axe/): A browser add-on for evaluating the accessibility of web pages within the developer console.
+* [Wave](http://wave.webaim.org/): A web tool and browser add-on for evaluating the accessibility of web pages.
+* [NVDA](http://www.nvda-project.org/): A screenreader for Windows (open source).
+* [JAWS](http://www.freedomscientific.com/products/fs/jaws-product-page.asp): The most popular screenreader. Available for Windows; costs money.
+* [VoiceOver](http://www.apple.com/accessibility/voiceover/): Built-in screenreader for Mac.
+* [VoiceOver for iOS](http://www.apple.com/accessibility/iphone/vision.html): Built-in screenreader for iOS.
+* [Chrome Vox](http://www.chromevox.com/): A screenreader for ChromeOS.
+* [WCAG 2.0 Cheat Sheet](http://www.w3.org/2009/cheatsheet/#wcag2) A simplified look at WCAG 2.0.
+* [WCAG 2.0 Mind Map](http://stamfordinteractive.com.au/wp-content/uploads/2013/03/WCAG20Map.pdf): A clickable mind-map that helps put the web accessibility guidelines into perspective. This WCAG 2.0-on-a-single-page reference is an alternative way to view the guidelines.
```
Comment (ryelle) [inline docs/accessibility.md]: This doesn't exist anymore 🙁

File: docs/accessibility.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+* [An Alt Text Decision Tree](http://dev.w3.org/html5/alt-techniques/developer.html#tree): A decision tree for deciding when and how to implement alt text in your work on the web. This is a work in progress by the editors of the HTML5 spec, but it's extremely useful in its current form.
+
+## Utilities in Calypso
+
+As we work to make Calypso more accessible, we'll probably add more things here.
+
+* [accessible-focus](client/lib/accessible-focus/README.md): A small module which is run at client startup and adds an `accessible-focus` class to the document's html element when keyboard navigation is detected, so that obvious focus styles can be added without being distracting for non-keyboard users.
```
Comment (ryelle) [inline docs/accessibility.md]: We also have [`Focusable`](https://github.com/Automattic/wp-calypso/tree/master/client/components/focusable) & [`ScreenReaderText`](https://github.com/Automattic/wp-calypso/tree/master/client/components/screen-reader-text) now.

File: docs/accessibility.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+* [Simply Accessible](http://simplyaccessible.com/): A blog with in-depth web accessibility tutorials.
+* [WebAxe](http://www.webaxe.org/): A blog and podcast about web accessibility.
+* [456 Berea Street](http://www.456bereastreet.com/): A blog on accessibility and web development by Roger Johansson.
+* [Accessible Culture](http://www.accessibleculture.org/): A blog about web accessibility by Jason Kiss, an accessibility and standards researcher located in Wellington, New Zealand.
+
+## Tools
+
+Find tools that will help you bring accessibility into your workflow.
+
+### UX and Design
```
Comment (ryelle) [inline docs/accessibility.md]: a "thinking out loud" note - it would be great to see a designer/design team make a page like the [typography page](https://wpcalypso.wordpress.com/devdocs/typography) or [the design handbook colors page](https://wordpress.com/design-handbook/colors/), but up to date & with indication of which colors can be used as text color against which background.

File: docs/accessibility-checklist.md [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+
+## Screen Reader Support
+
+* Do all form controls have associated labels and ARIA tags?
+* Are related form fields grouped with `fieldset` and described with `legend` tags?
+* Do all images have [descriptive alt text](https://a11yproject.com/posts/alt-text/)?
+* Do any visually-hidden elements have `aria-hidden` tags to hide them from screen readers?
+* Do all links have descriptive text (saying where they go as opposed to something like "click here")?
+* Do any modals or error messages provide useful feedback (via `aria-live` or similar tags)?
+* Give it a once-over using the [aXe browser extension](https://www.deque.com/products/axe/) to spot any obvious issues.
```
Comment (ryelle) [inline docs/accessibility-checklist.md]: Maybe move this up to the general section, since most of the things it catches on Calypso are contrast-related 🙂

File: docs/accessibility.md (hunk 6 above)
Comment (codebykat) [reply docs/accessibility.md]: I'd rather have less information and make sure it's cherrypicked, no one's going to read 200 links

File: docs/accessibility.md (hunk 8 above)
Comment (codebykat) [reply docs/accessibility.md]: `ScreenReaderText` doesn't have a README, but I think I figured it out

**Comment by codebykat** [comment]
@ryelle I think I addressed everything, but please do feel free to take it over if you'd like to refine :)

**Comment by ryelle** [comment]
This looks great, thanks for revitalizing it (& taking all my feedback). I can't approve the PR since I started it 😆- but I think it's good to go.

File: docs/accessibility-checklist.md [hunk 11]
```diff
@@ -0,0 +1,38 @@
+# Accessibility Checklist
+
+This document aims to provide a quick testing workflow that will help you find and fix common accessibility issues, whether you're writing a PR or reviewing.
+
+For more background and resources, see [Accessibility](accessibility.md).
+
+## Testing Tools
+
+* Do all `js-a11y` linting rules pass?
```
Comment (ockham) [inline docs/accessibility-checklist.md]: Super-tiny drive-by nitpick 😬: `jsx-a11y`

File: docs/accessibility.md (hunk 9 above)
Comment (gwwar) [reply docs/accessibility.md]: cc @megsfulton for the idea

**Comment by codebykat** [comment]
Thanks @ockham! Nitpick fixed 😄  I'll leave this unmerged until after the weekend to make sure everyone who wants to take a look, has had a chance to do so.

File: docs/accessibility.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+* [W3C’s Web Accessibility Initiative](http://www.w3.org/WAI/): The World Wide Web Consortium's Web Accessibility Initiative provides strategies, guidelines and resources to make the Web accessible to people with disabilities.
+* [The Six Simplest Accessibility Tests Anyone Can Do](http://www.karlgroves.com/2013/09/05/the-6-simplest-web-accessibility-tests-anyone-can-do/): A handful of simple things you can do to think about and implement accessibility with little effort as you work on projects.
+* [Intro to Cognitive Accessibility](jkg3.com/Journal/cognitive-accessibility-101-part-1-what-is-cognitive-accessibility)
```
Comment (gwwar) [inline docs/accessibility.md]: In the rich text preview, this was redirecting to github, might be safer to use the add the http:// protocol prefix here.

Comment (codebykat) [reply docs/accessibility.md]: Good catch, fixed 👍
